### ISSUE TITLE
No mention of certain features' limitations after using find in the UG

### ISSUE THREAD

**Issue by maj0-0** [maj0-0/pe#9]
Under the `find` feature in the UG, it would be helpful to include that the `list` command needs to be used before trying to edit contacts outside of the result given by `find`. Otherwise, it just shows that the index is not found.

![image.png](https://raw.githubusercontent.com/maj0-0/pe/main/files/44d891a4-00f8-4bfa-9f37-d39fcc4ef7dc.png)

I think it's fine that editting only works for contacts in the student panel, since the index is used for it, but it would be helpful to mention it in the UG, as the commands like insurance, addappt etc. are all affected by this.

**Comment by nus-se-script** [comment]
# Team's Response
Hi there, thanks for bringing up this issue.

From our understanding, what you meant here is that, as our find command will reduce the displayed list to a certain subset of customers, the user would then not be able to use edit(or any other commands that work on indices) on customers not in the list.

We have structured our response as follows:
> Under the find feature in the UG, it would be helpful to include that the list command needs to be used before trying to edit contacts outside of the result given by find. 

We do not quite agree with this, as we think that the find feature is not necessarily followed up by the edit command(or any other commands that work on indices), the find command itself is used to find customers with certain attributes. Hence, we think that mentioning of follow up actions is not really appropriate to be put under find command, they are separate actions to be performed by users.

Rather, we think that the restriction of the indices that could be used should be mentioned under the command that requires the indices itself, which is what we did in our UG, as mentioned by you too as follows, where the `some other commands` you mentioned is all commands that work on indices.
> since some commands mention that " should only be one of the indices shown in the displayed list"

As a conclusion, the main reason we are rejecting this bug is that the find and edit (or any other commands that work on indices) are two separate features, and the indices restrictions is bonded to the edit command (or any other commands that work on indices), not the find command, hence it should not be mentioned in the find section.
# Items for the Tester to Verify
## :question: Issue response

Team chose [`response.Rejected`]

- [x] I disagree

**Reason for disagreement:** Thank you for your detailed response regarding the interaction between the find and edit commands in our User Guide (UG). I understand and appreciate your perspective that the find and edit commands are separate features and that their functionalities are distinct, which is why labeled it as a **Documentation Bug**, and not a feature flaw, or functionality bug. I do not disagree with your implementation of find. However, I'd like to gently elaborate on my earlier point, considering the user experience and potential for confusion.

While I agree that the find feature isn't necessarily followed by the edit command, I do not think it complete rules out the possibility, users often perform a sequence of actions where understanding the impact of one command on another is crucial. In this case, the find command changes the context by altering the displayed list, which directly affects how subsequent commands operate. Highlighting this effect within the find feature's documentation could provide valuable context to users, especially when they attempt to use commands like edit on customers not currently displayed.

Including a brief note under the find command about the need to use the list command to revert to the full list before editing other contacts could be very helpful. This clarification would ensure users are fully aware of why certain indices might not work after a find operation. It's about enhancing the user's understanding of how these commands interact with each other.

As you rightly pointed out, mentioning "should only be one of the indices shown in the displayed list" under **other** commands, is in fact helpful. However, again, I do not think that is sufficient, as there is no other mention of what could cause this or what this implies. And based on your features the only thing that could cause this is the `find` feature. Additionally, if someone were just referring to the find command, and tried to find then tried to edit another customer using the index that showed up previously in the list, they would not be able to and there is no indication under the find feature on why this happens. They'd need to check other features in the UG to understand the issue.

In conclusion, I think since the updating of the displayed list occurs due to the find feature, it should be included as a remark in the find feature section of the UG. This small addition could significantly improve the usability and clarity of the guide, particularly for new users or those not as familiar with the interplay between different commands.


<catcher-end-of-segment><hr>
